usecases/replica/remote_incoming.go

```diff
@@ -13,119 +13,190 @@ package replica
+	if simpleResp != nil {
+		return *simpleResp
+	}
+	return index.CommitReplication(shardName, requestID)
 }
 
 func (rri *RemoteReplicaIncoming) AbortReplication(indexName,
 	shardName, requestID string,
 ) interface{} {
-	return rri.repo.AbortReplication(indexName, shardName, requestID)
+	index, simpleResp := rri.indexForIncomingRead(context.Background(), indexName)
+	if simpleResp != nil {
+		return *simpleResp
+	}
+	return index.AbortReplication(shardName, requestID)
 }
 
 func (rri *RemoteReplicaIncoming) OverwriteObjects(ctx context.Context,
 	indexName, shardName string, vobjects []*objects.VObject,
 ) ([]RepairResponse, error) {
-	return rri.repo.OverwriteObjects(ctx, indexName, shardName, vobjects)
+	index, simpleResp := rri.indexForIncomingRead(ctx, indexName)
+	if simpleResp != nil {
+		return nil, simpleResp.Errors[0].Err
+	}
+	return index.OverwriteObjects(ctx, shardName, vobjects)
 }
 
 func (rri *RemoteReplicaIncoming) FetchObject(ctx context.Context,
 	indexName, shardName string, id strfmt.UUID,
 ) (objects.Replica, error) {
-	return rri.repo.FetchObject(ctx, indexName, shardName, id)
+	index, simpleResp := rri.indexForIncomingRead(ctx, indexName)
+	if simpleResp != nil {
+		return objects.Replica{}, simpleResp.Errors[0].Err
+	}
+	return index.FetchObject(ctx, shardName, id)
 }
 
 func (rri *RemoteReplicaIncoming) FetchObjects(ctx context.Context,
 	indexName, shardName string, ids []strfmt.UUID,
 ) ([]objects.Replica, error) {
-	return rri.repo.FetchObjects(ctx, indexName, shardName, ids)
+	index, simpleResp := rri.indexForIncomingRead(ctx, indexName)
+	if simpleResp != nil {
+		return []objects.Replica{}, simpleResp.Errors[0].Err
+	}
+	return index.FetchObjects(ctx, shardName, ids)
 }
 
 func (rri *RemoteReplicaIncoming) DigestObjects(ctx context.Context,
 	indexName, shardName string, ids []strfmt.UUID,
 ) (result []RepairResponse, err error) {
-	return rri.repo.DigestObjects(ctx, indexName, shardName, ids)
+	index, simpleResp := rri.indexForIncomingRead(ctx, indexName)
+	if simpleResp != nil {
+		return []RepairResponse{}, simpleResp.Errors[0].Err
+	}
+	return index.DigestObjects(ctx, shardName, ids)
+}
+
+func (rri *RemoteReplicaIncoming) indexForIncomingRead(ctx context.Context, indexName string) (RemoteIndexIncomingRepo, *SimpleResponse) {
+	index := rri.repo.GetIndexForIncomingReplica(schema.ClassName(indexName))
+	if index == nil {
+		return nil, &SimpleResponse{Errors: []Error{{Err: fmt.Errorf("local index %q not found", indexName)}}}
+	}
+	return index, nil
+}
+
+func (rri *RemoteReplicaIncoming) indexForIncomingWrite(ctx context.Context, indexName string,
+	schemaVersion uint64,
+) (RemoteIndexIncomingRepo, *SimpleResponse) {
+	// wait for schema and store to reach version >= schemaVersion
+	if _, err := rri.schema.ReadOnlyClassWithVersion(ctx, indexName, schemaVersion); err != nil {
+		return nil, &SimpleResponse{Errors: []Error{{Err: fmt.Errorf("could not find class %q in schema: %w", indexName, err)}}}
```

Comment:
**suggestion (if-minor):**  `ReadOnlyClassWithVersion` doesn't bubble up the err if the class not found can we add also if `class == nil `  if we do wanna continue also after maybe then remove checking error to avoid any assumption.

Reply:
It's a good suggestion, the one thing I'm not sure is the relation between class and index. I'm not 100% if we can have a class absent, yet an index present in which we want to add an object. As we're actually retrieving the index below and just waiting on the schema version using ReadOnlyClassVersion i'd say we should act on the index missing not the class (which we do). Let me know your thoughts.

Reply:
i'd suggest then we keep it as it's to not change any beahviour, if you'd like to add `TODO-RAFT` for us to refactor that part and make sure it's not misinterpreted